internal/cmdrender/cmdrender.go

```diff
@@ -50,6 +50,8 @@ func NewRunner(ctx context.Context, parent string) *Runner {
 		fmt.Sprintf("output resources are written to provided location. Allowed values: %s|%s|<OUT_DIR_PATH>", cmdutil.Stdout, cmdutil.Unwrap))
 	c.Flags().StringVar(&r.imagePullPolicy, "image-pull-policy", string(fnruntime.IfNotPresentPull),
 		fmt.Sprintf("pull image before running the container. It must be one of %s, %s and %s.", fnruntime.AlwaysPull, fnruntime.IfNotPresentPull, fnruntime.NeverPull))
+	c.Flags().StringVar(&r.runtime, "runtime", string(fnruntime.DockerRuntime),
```

Comment:
I am still thinking if we should use environment variable to control this. One advantage I see is: package users will always use `kpt fn render` without worrying about the runtimes and platform team can control the function runner by changing the environment variable without changing the `kpt fn` commands. It sort of decouples the team managing the runtime environments.

Reply:
I agree with your reasoning about env var is a better solution here.

I just realized that completely relying env var makes parallel testing impossible in golang. See: https://pkg.go.dev/testing#T.Setenv and https://stackoverflow.com/questions/58356031/how-to-run-test-in-parallel-requiring-different-values-for-env-variables

I see 2 options:
- Make all e2e test sequential. E2e tests will take longer to run.
- Support both env var and flag. And we use flag in the e2e tests.

@droot WDYT?

Reply:
That's a good point.

I think supporting both env var/flag could get confusing for the users, so going with an explicit command-line flag is probably ok. So, let's keep it as is then.